packages/astro/src/core/ssr/resolve-config-paths.ts

```diff
@@ -0,0 +1,94 @@
+import { tsconfigResolverSync, TsConfigResultSuccess } from 'tsconfig-resolver';
+import fs from 'fs'
+import { posix as ps } from 'path'
+import type * as vite from 'vite'
+
+/** Return the nearest tsconfig.json or jsconfig.json configuration result. */
+const getConfigResult = (): TsConfigResultSuccess | null => {
+	/** Closest tsconfig.json */
+	const tsConfig = tsconfigResolverSync({ searchName: 'tsconfig.json' })
+
+	// return tsconfig.json, if it exists
+	if (tsConfig.exists) return tsConfig
+
+	/** Closest jsconfig.json */
+	const jsConfig = tsconfigResolverSync({ searchName: 'jsconfig.json' })
+
+	// return jsconfig.json, if it exists
+	if (jsConfig.exists) return jsConfig
+
+	// return null if no config exists
+	return null
+}
+
+/** Return the resolvers from the tsconfig.json or jsconfig.json configuration. */
+const getConfigResolver = (): { baseUrl: string, paths: { [alias: string]: string[] } } | null => {
+	/** Configuration result */
+	const configResult = getConfigResult()
+
+	if (configResult && configResult.config.compilerOptions?.baseUrl) {
+		let { baseUrl, paths } = configResult.config.compilerOptions as {
+			baseUrl: string
+			paths: {
+				[key: string]: string[];
+			}
+		}
+
+		// resolve the base url from the configuration file directory
+		baseUrl = ps.resolve(ps.dirname(configResult.path), baseUrl)
+
+		// guarantee paths is an object
+		paths = Object(paths)
+
+		// sanitize and resolve all aliases from the configuration file directory
+		for (let [ alias, values ] of Object.entries(paths)) {
+			delete paths[alias]
+
+			paths[alias.replace(/\*$/, '')] = values = [].concat(values as any).map(
+				(value: string) => ps.resolve(baseUrl, value).replace(/\*$/, '')
+			)
+		}
+
+		return { baseUrl, paths }
+	}
+
+	return null
+}
+
+export default function resolveConfigPaths(): vite.PluginOption {
+	/** Resolver from the tsconfig.json or jsconfig.json configuration. */
+	const resolver = getConfigResolver()
+
+	if (!resolver) return {} as vite.PluginOption
+
+	return {
+		name: 'resolve-config-paths',
+		enforce: 'pre',
+		resolveId(source: string): string | null {
+			// conditionally resolve from each alias
+			for (let [ alias, values ] of Object.entries(resolver.paths)) {
+				if (source.startsWith(alias)) {
+					for (const value of values) {
+						const resolvedSource = source.replace(alias, value)
+
+						if (fs.existsSync(resolvedSource)) {
+							return resolvedSource
+						}
+					}
+				}
+			}
+
+			// otherwise, conditionally resolve from the baseUrl
+
+			const resolvedSource = ps.resolve(resolver.baseUrl, source)
```

Comment:
Rethinking the `resolveId` approach entirely, what if this plugin didn't do any resolution and just set the relevant `alias` option in Vite using [`config`](https://vitejs.dev/guide/api-plugin.html#config)?

Reply:
I’m open to this change. Would you give this a thumbs up if you feel strongly about it, and I’ll know to work on refactoring this?

Reply:
Ultimately, I went with the `this.resolve` strategy you pasted in the other comment. Done in dd26e2e34b703849baeaf3d5c60e6d3da175d54b